File: doc/codes/python/examples/api/compute/pattern/vortex.py
import numpy as np
from autd3.geometry import Autd3, Geometry
from autd3.units import m, s
from autd3_pattern import vortex, wavelength as calc_wavelength

geometry = Geometry([Autd3([0.0, 0.0, 0.0], [1.0, 0.0, 0.0, 0.0])])

target = geometry.center() + np.array([0.0, 0.0, 150.0])
axis = np.array([0.0, 0.0, 1.0])
order = 1
wavelength = calc_wavelength(340 * m / s)
dst = geometry.pattern_buffer()

# ANCHOR: api
vortex(geometry, target, axis, order, wavelength, dst)
# ANCHOR_END: api

File: doc/codes/csharp/examples/api/compute/pattern/vortex.cs
using System.Numerics;
using AUTD3;
using static AUTD3.Units;

namespace DocSamples.ApiComputePatternVortex;

internal static class Sample
{
    internal static void Run()
    {
        var geometry = new Geometry(new[] { new Autd3(Vector3.Zero) });

        var target = geometry.Center + new Vector3(0.0f, 0.0f, 150.0f);
        var axis = Vector3.UnitZ;
        var order = 1;
        var wavelength = Pattern.Wavelength(340.0f * m / s);
        var dst = geometry.PatternBuffer();

        // ANCHOR: api
        Pattern.Vortex(geometry, target, axis, order, wavelength, dst);
        // ANCHOR_END: api
    }
}
